metadata-ingestion/src/datahub/ingestion/transformer/base_transformer.py

```diff
@@ -180,6 +195,33 @@ def _transform_or_record_mcpw(
             self._record_mcp(envelope.record)
         return envelope if envelope.record.aspect is not None else None
 
+    def _handle_end_of_stream(
+        self, envelope: RecordEnvelope, entity_urn: str
+    ) -> Iterable[RecordEnvelope]:
+
+        if not isinstance(self, SingleAspectTransformer):
```

Comment:
let's make this work for both types of transformers

Reply:
done